File: Coupon.Admin/ClientApp/src/app/modules/products/dto/product.dto.ts
import { ImageDto } from "src/app/shared/data/image/image.dto";
import { ProductState } from "./product-state";

export interface ProductDto{
    id: number,
    title: string,
    fullTitle: string,
    validFrom: string,
    validUntil: string,
    conditions: string,
    mainImageId: string,
    mainImage: ImageDto,
    images: ImageDto[],
    providerId: string,
    provider: ProviderDto,
    oldPrice: string,
    newPrice: string,
    startAvailableCount: number,
    availableCount: number,
    onSale: boolean,
    state: ProductState
}

File: Coupon.Dto/ProductDto.cs
using Coupon.Common.Enums;
using Coupon.Data.Model;
using System;
using System.Collections.Generic;

namespace Coupon.Dto
{
    public class ProductDto
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string FullTitle { get; set; }

        public DateTimeOffset ValidFrom { get; set; }

        public DateTimeOffset ValidUntil { get; set; }

        public string Description { get; set; }

        public string Conditions { get; set; }

        public Guid MainImageId { get; set; }

        public Images MainImage { get; set; }

        public IEnumerable<Images> Images { get; set; }

        public Guid ProviderId { get; set; }

        public Providers Provider { get; set; }

        public decimal OldPrice { get; set; }

        public decimal NewPrice { get; set; }

        public int StartAvailableCount { get; set; }

        public int AvailableCount { get; set; }

        public bool OnSale { get; set; }

        public ProductState State { get; set; }
    }
}
